R/manifest-convert.R

```diff
@@ -0,0 +1,81 @@
+#' Convert manifest.json to an renv.lock file
+#'
+#' Use `rsc_manifest_to_renv_lock` to convert a `manifest.json` file from
+#' an RStudio Connect content bundle into an `renv.lock` file.
+#'
+#' This function can be useful when you need to recreate the package environment
+#' of a piece of content that is deployed to RStudio Connect. The content bundle
+#' contains a `manifest.json` file that is used to recreate the package environment.
+#' This function will let you convert that manifest file to an `renv.lock` file.
+#' Run `renv::restore()` after you've converted the file to restore the package
+#' environment.
+#'
+#' @param manifest The path to a manifest.json file.
+#' @param lockfile The path to a lockfile you want to update, can be an empty text file. Defaults to NA.
+#'
+#' @details
+#'
+#' By default the `lockfile` argument is set to `NA`. This will not create a new renv.lock file.
+#' Rather, it will return a lockfile object (see `?lockfile`) that can be used to create a new
+#' renv.lock file. If `lockfile` is set to a character string, a new file will be created with that
+#' path—e.g. `renv.lock`—and the lockfile object will be returned.
+#'
+#'
+#' @export
+rsc_manifest_to_renv_lock <- function(manifest = NULL, lockfile = NA) {
+
+  if (missing(manifest)) {
+    stop("Provide the path the manifest.json file.", call. = FALSE)
+  }
+
+  # create lockfile if non-existant and lockfile != NA
+  if (!file.exists(lockfile %&&% "") & !is.na(lockfile)) {
+    fmt <- "%s doesn't exist. Creating new file."
+    vwritef(fmt, lockfile)
+    file.create(lockfile)
+  }
+
+  cat("* Reading manifest.json...\n")
```

Comment:
Use `vwritef()` (if you think the user should be able to disable logging) or `writef()` instead.